### PR TITLE
Add documentation page generated from the cmd/*.rs. Close #223

### LINKED ISSUE #223: Generate examples.md from the doc comments

### PR DESCRIPTION
**Pull request by hikalium**
People often forget about updating the documentation. To prevent that, move the command-line level documentation to the beginning of the each command line interface sources and generate a searchable command line reference from it.

### CODE REVIEW

**Comment by mhiramat** [comment]
Can you update the document by replacing `lium` with `cro3`?

**Comment by hikalium** [comment]
@mhiramat updated all `lium` to `cro3`. Please take a look!

File: src/dut.rs [hunk 1]
```diff
@@ -417,7 +417,7 @@ impl SshInfo {
         // > Support for <zone_id> is intentionally omitted.
         // So that we can not parse dut string like:
         // > [fe80::8a54:1fff:fe0f:72a5%en0]
-        // TODO(hikalium): Impl workaround, possibly fix in the url crate.
+        // TODO(hikacro3): Impl workaround, possibly fix in the url crate.
```
Comment (HidenoriKobayashi) [inline src/dut.rs:420]: are you sure? 😄

Comment (hikalium) [reply src/dut.rs:420]: aaaaaaahhhh, good catch, thanks... (I'm not going to change my name to hikacro3 😂 )

File: docs/cmdline_preface.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+> [!NOTE]
+> This file is automatically generated by `make`.
+> Please modify src/cmd/*.rs files and rebuild cros to update.
```
Comment (HidenoriKobayashi) [inline docs/cmdline_preface.md:3]: Did you mean `rebuild cro3`?

Besides, I'm wondering who is the intended reader of this NOTE. Maybe something like "If you have modified src/cmd/*.rs files, rebuild cro3 to update this page"?

**Review by HidenoriKobayashi — COMMENTED** [review]
I just checked the rendered result directly.  I have no objection to this change as one of the steps to improve the document.

I think this PR needs a review from more than one person. (CR+1 from me)

File: docs/cmdline_preface.md (hunk 2 above)
Comment (hikalium) [reply docs/cmdline_preface.md]: Ah, good catch... Fixed! Also, updated the text to make it better. Thank you!

**Comment by hikalium** [comment]
@mhiramat Please review and CR+2 if it seems to be ok...!

**Review by mhiramat — APPROVED** [review]
LGTM, thanks!

**Pull request merged**